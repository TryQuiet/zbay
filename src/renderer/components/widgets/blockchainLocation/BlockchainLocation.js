import React from 'react'
import PropTypes from 'prop-types'

import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core/styles'

import Modal from '../../ui/Modal'

const styles = theme => ({
  root: {
    marginTop: -60,
    backgroundColor: theme.palette.colors.white,
    border: 'none'
  },
  button: {
    marginTop: 20,
    height: 70,
    width: 300,
    fontSize: '0.9rem',
    backgroundColor: theme.palette.colors.zbayBlue
  },
  link: {
    textAlign: 'center',
    marginTop: 20,
    width: 300,
    marginLeft: 2,
    fontSize: '14px',
    cursor: 'pointer',
    color: theme.palette.colors.linkBlue
  },
  title: {
    marginBottom: 16
  },
  bold: {
    fontWeight: 'bold'
  },
  text: {
    textAlign: 'justify',
    textAlignLast: 'center'
  }
})

export const BlockchainLocationModal = ({ classes, open, handleSelection }) => {
  return (
    <Modal open={open} isCloseDisabled>
      <Grid container direction='column' className={classes.root} alignContent='center' justify='center'>
        <Grid container item justify='center'>
          <Grid item className={classes.title}>
            <Typography variant='h3'>Use existing Zcash install?</Typography>
          </Grid>
        </Grid>
        <Grid container item justify='center'>
          <Grid item>
            <Typography variant='body2' className={classes.text}>Oh awesome! It looks like you may have already installed Zcash. Zbay includes Zcash but we can use your existing Zcash instead -
              <span className={classes.bold}> IF it's a full node (not a light wallet) with default settings. </span>Not sure? Use ours! It should sync quickly.
            </Typography>
          </Grid>
        </Grid>
        <Grid container justify='center' alignItems={'flex-start'}>
          <Grid item container justify={'center'} xs={12}>
            <Button
              variant='contained'
              size='large'
              color='primary'
              type='submit'
              onClick={() => handleSelection('DOWNLOAD_NEW')}
              fullWidth
              className={classes.button}
            >
        Install Zcash again (recommended)
            </Button>
          </Grid>
          <Grid item container justify={'center'} xs={12}>
            <Typography className={classes.link} variant={'caption'} onClick={() => handleSelection('EXISTING')}>{`Use my existing Zcash install (advanced)`}</Typography>
          </Grid>
        </Grid>
      </Grid>
    </Modal>
  )
}

BlockchainLocationModal.propTypes = {
  open: PropTypes.bool.isRequired,
  handleSelection: PropTypes.func.isRequired
}

export default withStyles(styles)(BlockchainLocationModal)
