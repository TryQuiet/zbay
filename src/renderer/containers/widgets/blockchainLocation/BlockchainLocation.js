import * as R from 'ramda'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import BlockchainLocationModal from '../../../components/widgets/blockchainLocation/BlockchainLocation'
import appHandlers from '../../../store/handlers/app'
import { withModal } from '../../../store/handlers/modals'

export const mapDispatchToProps = dispatch => bindActionCreators({
  handleSelection: (userChoice) => appHandlers.epics.proceedWithSyncing(userChoice)
}, dispatch)

export default R.compose(
  connect(null, mapDispatchToProps),
  withModal('blockchainLocation')
)(BlockchainLocationModal)
