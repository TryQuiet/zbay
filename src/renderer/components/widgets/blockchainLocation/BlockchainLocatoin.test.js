/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { BlockchainLocationModal } from './BlockchainLocation'
import { mockClasses } from '../../../../shared/testing/mocks'

describe('BlockchainLocationModal', () => {
  it('renders component', () => {
    const result = shallow(
      <BlockchainLocationModal
        open
        classes={mockClasses}
        handleSelection={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
