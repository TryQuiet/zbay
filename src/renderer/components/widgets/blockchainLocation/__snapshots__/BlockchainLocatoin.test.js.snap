// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockchainLocationModal renders component 1`] = `
<Memo(WithStyles(Modal))
  isCloseDisabled={true}
  open={true}
>
  <WithStyles(ForwardRef(Grid))
    alignContent="center"
    className="root-mock"
    container={true}
    direction="column"
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        className="title-mock"
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          Use existing Zcash install?
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="text-mock"
          variant="body2"
        >
          Oh awesome! It looks like you may have already installed Zcash. Zbay includes Zcash but we can use your existing Zcash instead -
          <span
            className="bold-mock"
          >
             IF it's a full node (not a light wallet) with default settings. 
          </span>
          Not sure? Use ours! It should sync quickly.
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-start"
      container={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        item={true}
        justify="center"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          className="button-mock"
          color="primary"
          fullWidth={true}
          onClick={[Function]}
          size="large"
          type="submit"
          variant="contained"
        >
          Install Zcash again (recommended)
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        item={true}
        justify="center"
        xs={12}
      >
        <WithStyles(ForwardRef(Typography))
          className="link-mock"
          onClick={[Function]}
          variant="caption"
        >
          Use my existing Zcash install (advanced)
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(Modal))>
`;
